### PR TITLE
Fixed empty value when get variables out of collectd hook.

### PR DESCRIPTION
**Pull request by charz**

### CODE REVIEW

File: collectd_plugin/collectd_plugin.py [hunk 1]
```diff
@@ -956,7 +962,5 @@ def _should_filter(values, data, ds):
     data = {}
     collectd.register_init(init)
     collectd.register_config(config, data)
-    if data.get("monitor_endpoint") or (
-            data.get("carbon_host") and data.get("carbon_port")):
```
Comment (clayg) [inline collectd_plugin/collectd_plugin.py:960]: in the config function it seems like we could have a valid writer thread/queue even if we've *only* configured carbon_host/port and not the monitor_endpoint (I don't know if that was a legacy configuration?)

**Review by clayg — COMMENTED** [review]
I can't find anything in the collectd python plugin docs that suggest we HAVE to register_write/read ???

https://collectd.org/documentation/manpages/collectd-python.5.shtml
https://github.com/signalfx/collectd-example/blob/master/example_plugin.py

I got the sense maybe controller config missing monitor_endpoint broke the controller metrics - but we fixed that... do you know if the controller config had carbon_host/carbon_port?  Wouldn't this change have made the read/write on controller a noop?  Is there something else we're trying to do here?

File: collectd_plugin/collectd_plugin.py (hunk 1 above)
Comment (charz) [reply collectd_plugin/collectd_plugin.py:960]: >I got the sense maybe controller config missing monitor_endpoint broke the controller metrics - but we fixed that... 

We do have this issue on nodes, if you check the the monitor page for the node. All are empty once we upgrade to new collectd-plugin.

> carbon_host/port and not the monitor_endpoint 

No, I *never* see `carbon_host/port` in collectd configuration. 

I tried to log `_log("endpoint: %s", data.get("monitor_endpoint"))` or `_log("data: %s", data)` in here and  got nothing.

Comment (charz) [reply collectd_plugin/collectd_plugin.py:960]: >I can't find anything in the collectd python plugin docs that suggest we HAVE to register_write/read ???
> https://collectd.org/documentation/manpages/collectd-python.5.shtml
> https://github.com/signalfx/collectd-example/blob/master/example_plugin.py

I got the samething, and not very clear why I have `monitor_endpoint` in plug-in section, but not able to get these variable in main (outsidf of collectd.regist_XXX fucntions).

File: collectd_plugin/collectd_plugin.py [hunk 2]
```diff
@@ -956,7 +962,5 @@ def _should_filter(values, data, ds):
     data = {}
```
Comment (clayg) [inline collectd_plugin/collectd_plugin.py:962]: can confirm: during module import this is empty

registering the config handler, isn't the same as *calling* the config handler - so data is still empty.

File: collectd_plugin/collectd_plugin.py [hunk 3]
```diff
@@ -397,6 +400,9 @@ def read(data):
     correctness.
     """
 
+    if not data.get("monitor_endpoint"):
+         return
```
Comment (clayg) [inline collectd_plugin/collectd_plugin.py:404]: it looks like we always configure `data['metrics_queue']` so this is probably what we want

**Review by clayg — APPROVED** [review]
yes

**Pull request merged**